### PR TITLE
Add more Remote Migration Integration Tests 

### PR DESCRIPTION
**Pull request by gbbafna**
### Description
Adds more remote migration integ tests 

1. Replica recovery 
2. Failover to docrep copy 
3. End to end testing with refresh/flush , covering primary-replica relocation 
4. Minor refactoring 

### Related Issues
Resolves none. 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [] ~Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))~
- [ ] ~Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteStoreMigrationTestCase.java [hunk 1]
```diff
@@ -71,4 +86,81 @@ public void testMigrationDirections() {
...
+    - Asserts all shards are migrated to remote store
+    - Asserts doc count across all shards
+    - Continuos indexing with refresh/flush happening
+     */
+    public void testEndToEndRemoteMigration() throws Exception {
+        internalCluster().setBootstrapClusterManagerNodeIndex(0);
+        setAddRemote(false);
+        List<String> docRepNodes = internalCluster().startNodes(2);
+        ClusterUpdateSettingsRequest updateSettingsRequest = new ClusterUpdateSettingsRequest();
+        updateSettingsRequest.persistentSettings(Settings.builder().put(REMOTE_STORE_COMPATIBILITY_MODE_SETTING.getKey(), "mixed"));
```
Comment (shourya035) [inline server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteStoreMigrationTestCase.java:104]: Do we need to update this twice? We can use the `initDocRepToRemoteMigration` method in `MigrationBaseTestCase` class directly after all docrep nodes are started

File: server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteReplicaRecoveryIT.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    protected int maximumNumberOfReplicas() {
+        return 1;
+    }
+
+    protected int minimumNumberOfReplicas() {
+        return 1;
+    }
+
+    @TestLogging(reason = "Getting trace logs from replication package", value = "org.opensearch.index.translog.RemoteFsTranslog:TRACE,org.opensearch.index.shard.RemoteStoreRefreshListener:TRACE,"
```
Comment (shourya035) [inline server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteReplicaRecoveryIT.java:53]: Let's remove this if not needed

File: server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteDualReplicationIT.java [hunk 3]
```diff
@@ -389,6 +389,129 @@ public void testFailoverRemotePrimaryToDocrepReplica() throws Exception {
...
+            }
+        }
+        int firstBatch = (int) asyncIndexingService.getIndexedDocs();
+        assertReplicaAndPrimaryConsistency(FAILOVER_REMOTE_TO_DOCREP, firstBatch, 0);
+
+        logger.info("---> Stop remote store enabled node");
+        internalCluster().stopRandomNode(InternalTestCluster.nameFilter(remoteNodeName));
+        ensureStableCluster(3);
+        ensureYellow(FAILOVER_REMOTE_TO_DOCREP);
+
```
Comment (shourya035) [inline server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteDualReplicationIT.java:490]: We can add an assertion that the primary shard copy is in a remote enabled node here

File: server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteDualReplicationIT.java [hunk 4]
```diff
@@ -389,6 +389,129 @@ public void testFailoverRemotePrimaryToDocrepReplica() throws Exception {
...
+        internalCluster().startClusterManagerOnlyNode();
+
+        logger.info("---> Starting 1 docrep data nodes");
+        String docrepNodeName = internalCluster().startDataOnlyNode();
+        internalCluster().validateClusterFormed();
+        assertEquals(internalCluster().client().admin().cluster().prepareGetRepositories().get().repositories().size(), 0);
+
+        logger.info("---> Creating index with 0 replica");
+        Settings excludeRemoteNode = Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0).build();
+        createIndex(FAILOVER_REMOTE_TO_DOCREP, excludeRemoteNode);
```
Comment (shourya035) [inline server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteDualReplicationIT.java:413]: nit: Can we change the index name to `FAILOVER_REMOTE_TO_REMOTE`?

File: server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteDualReplicationIT.java [hunk 5]
```diff
@@ -389,6 +389,129 @@ public void testFailoverRemotePrimaryToDocrepReplica() throws Exception {
         );
     }
 
+    /*
+    Scenario:
```
Comment (shourya035) [inline server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteDualReplicationIT.java:393]: We need to change the javadoc to reflect the 3rd Remote node we are adding

File: server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteStoreMigrationTestCase.java (hunk 1 above)
Comment (gbbafna) [reply server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteStoreMigrationTestCase.java:104]: done

**Review by kkmr — APPROVED** [review]

**Review by sachinpkale — APPROVED** [review]

**Pull request merged**